File: go/update_add_on_request.go
// this file is @generated
package meteroid

type UpdateAddOnRequest struct {
	Description *string `json:"description,omitempty"`

	MaxInstancesPerSubscription *int32 `json:"max_instances_per_subscription,omitempty"`

	Name *string `json:"name,omitempty"`

	PriceId *PriceId `json:"price_id,omitempty"`

	SelfServiceable *bool `json:"self_serviceable,omitempty"`
}

File: python/meteroid/models/update_add_on_request.py
# this file is @generated
import dataclasses
import typing as t

from ..serialization import BaseModel
from .price_id import PriceId


@dataclasses.dataclass
class UpdateAddOnRequest(BaseModel):
    description: t.Optional[str] = None

    max_instances_per_subscription: t.Optional[int] = None

    name: t.Optional[str] = None

    price_id: t.Optional[PriceId] = None

    self_serviceable: t.Optional[bool] = None
